src/renderers/shaders/ShaderChunk/lights_lambert_vertex.glsl

```diff
@@ -105,24 +64,26 @@ vec3 normal = normalize( transformedNormal );
 
 	for ( int i = 0; i < MAX_HEMI_LIGHTS; i ++ ) {
 
-		vec3 lightDir = hemisphereLightDirection[ i ];
-
-		// diffuse
+		IncidentLight indirectLight = getHemisphereIndirectLight( hemisphereLights[ i ], geometry );
 
-		float dotProduct = dot( normal, lightDir );
-
-		float hemiDiffuseWeight = 0.5 * dotProduct + 0.5;
-
-		vLightFront += mix( hemisphereLightGroundColor[ i ], hemisphereLightSkyColor[ i ], hemiDiffuseWeight );
+		BRDF_Lambert( indirectLight, geometry, diffuse, frontReflectedLight );
 
 		#ifdef DOUBLE_SIDED
+	
+			indirectLight = getHemisphereIndirectLight( hemisphereLights[ i ], backGeometry );
 
-			float hemiDiffuseWeightBack = - 0.5 * dotProduct + 0.5;
-
-			vLightBack += mix( hemisphereLightGroundColor[ i ], hemisphereLightSkyColor[ i ], hemiDiffuseWeightBack );
+			BRDF_Lambert( indirectLight, backGeometry, diffuse, backReflectedLight );
 
 		#endif
 
 	}
 
 #endif
+
+vLightFront = frontReflectedLight.diffuse;
```

Comment:
vLightFront now has a different definition since `diffuse` is multiplied into the formula.